src/Symfony/Component/Workflow/Marking.php

```diff
@@ -31,36 +35,43 @@ public function __construct(array $representation = [])
         }
     }
 
-    /**
-     * @return void
-     */
-    public function mark(string $place)
+    public function mark(string|\UnitEnum $place): void
```

Comment:
Note that this method is not final and could be overridden downstream. Widening the parameter type and narrowing the native return type are both breaking changes.

Reply:
I propose to make Marking.php the final class

Reply:
We cannot do that now, as making a class final is also a breaking change. So we can at most flag it as final using PHPdoc and make it final in 7.0. But that would also mean that we probably need to introduce a `MarkingInterface` and update the rest of the component to use that typehint, as otherwise this is fully closed for extension.

Reply:
Yes, you're right, it's better to leave it like that than to make many changes in favor of MarkingInterface